File: src/engine/commands/statscommand.cpp
#include "statscommand.h"

#include "race.h"
#include "realm.h"
#include "util.h"


#define super Command

StatsCommand::StatsCommand(QObject *parent) :
    super(parent) {

    setDescription("Display your character stats.\n"
                   "\n"
                   "Example: stats");
}

StatsCommand::~StatsCommand() {
}

void StatsCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    CharacterStats stats = player->stats();

    send("\n" +
         Util::colorize(QString("/===================== Overview ==================\\\n"
                                "| %1 |\n"
                                "\\=================================================/\n")
                        .arg(Util::center(QString("%1, a %2 %3")
                             .arg(player->name(),
                                  player->race().cast<Race *>()->adjective(),
                                  player->characterClass()->name()), 47)), Green));
    send(Util::colorize(QString(" /===== Physique ======\\   /==== Attributes =====\\ \n"
                                " | Gender:  %7     |   | STR:  %1   DEX:  %2 | \n"
                                " | Height:  %8      |   | VIT:  %3   END:  %4 | \n"
                                " | Weight:  %9      |   | INT:  %5   FAI:  %6 | \n"
                                " \\=====================/   \\=====================/ \n")
                        .arg(QString::number(stats.strength).rightJustified(2),
                             QString::number(stats.dexterity).rightJustified(2),
                             QString::number(stats.vitality).rightJustified(2),
                             QString::number(stats.endurance).rightJustified(2),
                             QString::number(stats.intelligence).rightJustified(2),
                             QString::number(stats.faith).rightJustified(2),
                             Util::capitalize(player->gender()).leftJustified(6),
                             Util::formatHeight(player->height()).leftJustified(5),
                             Util::formatWeight(player->weight()).leftJustified(5)),
                        Olive) +
         "\n");
}

File: data/commands/statscommand.js
function StatsCommand() {
    Command.call(this);

    this.setDescription("Display your character stats.\n" +
                        "\n" +
                        "Example: stats");
}

StatsCommand.prototype = new Command();
StatsCommand.prototype.constructor = StatsCommand;

StatsCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var stats = player.stats();

    player.send("\n" +
                Util.colorize("/===================== Overview ==================\\\n" +
                              "| %1 |\n".arg(Util.center("%1, a %2 %3"
                                                         .arg(player.name,
                                                              player.race.adjective,
                                                              player.characterClass.name))) +
                              "\\=================================================/\n",
                              Color.Green) +
                "\n" +
                Util.colorize(" /===== Physique ======\\   /==== Attributes =====\\ \n" +
                              " | Gender:  %1     |   | STR:  %2   DEX:  %3 | \n"
                              .arg(player.gender.leftJustified(6),
                                   stats[STRENGTH].toString().leftJustified(2),
                                   stats[DEXTERITY].toString().leftJustified(2)) +
                              " | Height:  %1      |   | VIT:  %2   END:  %3 | \n"
                              .arg(Util.formatHeight(player.height).leftJustified(5),
                                   stats[VITALITY].toString().leftJustified(2),
                                   stats[ENDURANCE].toString().leftJustified(2)) +
                              " | Weight:  %1      |   | INT:  %2   FAI:  %3 | \n"
                              .arg(Util.formatWeight(player.weight).leftJustified(5),
                                   stats[INTELLIGENCE].toString().leftJustified(2),
                                   stats[FAITH].toString().leftJustified(2)) +
                              " \\=====================/   \\=====================/ \n",
                              Color.Olive) +
             "\n");
};

CommandRegistry.registerCommand("stats", new StatsCommand());
